src/mod.ts

```diff
@@ -22,12 +23,86 @@ export const run = async function (functionDir: string) {
     return potentialFunctionFiles;
   });
 
-  // The CLI expects a JSON payload to be output to stdout
-  // This is formalized in the `run` hook of the CLI/SDK Tech Spec:
-  // https://corp.quip.com/0gDvAsqoaaYE/Proposal-CLI-SDK-Interface#temp:C:fOC1991c5aec8994d0db01d26260
-  console.log(JSON.stringify(resp || {}));
+  return JSON.stringify(resp || {});
+};
+
+// Start a http server that listens on the provided port
+// This server exposes two routes
+//  GET  `/health`    Returns a 200 OK
+//  POST `/functions` Accepts event payload and invokes `run`
+const startServer = async function (port: number) {
+  let server;
+  try {
+    server = Deno.listen({ port });
+  } catch (e) {
+    throw new Error(`Unable to listen on ${port}, got ${e}`);
+  }
+  for await (const conn of server) {
+    // In order to not be blocking, we need to handle each connection individually
+    // without awaiting the function
+    serveHttp(conn);
+  }
+
+  async function serveHttp(conn: Deno.Conn) {
+    // This "upgrades" a network connection into an HTTP connection.
+    const httpConn = Deno.serveHttp(conn);
+    // Each request sent over the HTTP connection will be yielded as an async
+    // iterator from the HTTP connection.
+    for await (const requestEvent of httpConn) {
+      // The native HTTP server uses the web standard `Request` and `Response`
+      // objects.
+
+      const url = new URL(requestEvent.request.url);
+      // A health check route
+      if (requestEvent.request.method == "GET" && url.pathname == "/health") {
+        return requestEvent.respondWith(
+          new Response("OK", {
+            status: 200,
+          }),
+        );
+      } else if (
+        requestEvent.request.method == "POST" && url.pathname == "/functions"
+      ) {
+        const body = await requestEvent.request.text();
+        try {
+          // run the user code
+          const response = await run(url.pathname.substring(1), body);
+          return requestEvent.respondWith(
+            new Response(response, {
+              status: 200,
+              headers: { "Content-Type": "application/json" },
+            }),
+          );
+        } catch (e) {
+          console.error(`Unable to run user supplied module caught error ${e}`);
+          requestEvent.respondWith(
```

Comment:
we should return here I think so we don't hit the default response below

Reply:
excellent point, I think I had it that way at some point then deleted the `return` will do